Lesson15/Busra/busra.js

```diff
@@ -0,0 +1,508 @@
+/*
+4. Extract Domain
+   - Define a function `getDomain(email)` that uses `slice` or `substring` to
+     extract everything after '@'.
+   - Log the domain to the console.
+   - Example: "user@example.com" -> "example.com"
+*/
+
+function getDomain(email){ 
+  const atIndex = email.indexOf('@');
+  const domain = email.slice(atIndex+1);
+  console.log(domain);
+}
+getDomain("busrademirkesen@gmail.com");
+
+/*
+5. Check Substring
+   - Define a function `containsWord(sentence, word)` that checks if the `sentence`
+     includes `word` (use the .includes() method).
+   - If true, log: "<word> found in sentence."
+   - Else, log: "<word> not found in sentence."
+*/
+
+function containsWord(sentence,word){
+  if(sentence.includes(word)){
+    console.log(word + " found in sentence");
+  }
+    else{
+      console.log(word + " is not found in sentence");
+    }
+  }
+
+
+containsWord("I need to study","study");
+
+/*
+6. File Extension Check
+   - Define a function `checkFileExtension(filename)` that checks if the filename
+     ends with ".pdf" using .endsWith().
+   - If it does, log: "This is a PDF file."
+   - Otherwise, log: "Not a PDF file."
+*/
+function checkFileExtension(filename){
+  if(filename.endsWith(".pdf")){
+    console.log("This is a PDF file.");
+  }
+  else{
+    console.log("Not a PDF file.");
+  }
+}
+checkFileExtension("dosya.pdf");
+checkFileExtension("dosya.jpeg");
+
+
+/*
+8. Palindrome Check
+   - Define a function `isPalindrome(str)` that checks if `str` is the same
+     forwards and backwards (normalize the string and compare it to reversed version of itself).
+   - If it is, log: "<str> is a palindrome"
+   - Otherwise, log: "<str> is not a palindrome"
+*/
+function isPalindrome(str) {
+  
+  let normalizeStr = str.toLowerCase().replace(/[^a-z0-9]/g, "");// büyük harf ve karakterli stringi normalize eder
+
+
+  let ters = normalizeStr.split('').reverse().join('');
```

Comment:
Practice using english names in programming. Feel free to ask copilot for help with naming.
```suggestion
  let reversedString = normalizeStr.split('').reverse().join('');
```